R/lm_lin.R

```diff
@@ -101,16 +119,45 @@ lm_lin <- function(formula,
   original_covar_names <- colnames(demeaned_covars)
 
   # Change name of centered covariates to end in bar
-  colnames(demeaned_covars) <- paste0(colnames(demeaned_covars), '_bar')
+  colnames(demeaned_covars) <- paste0(colnames(demeaned_covars), "_bar")
+
+  n_treat_cols <- ncol(treatment)
+  n_covars <- ncol(demeaned_covars)
 
   # Interacted
-  interacted_covars <- treatment * demeaned_covars
-  colnames(interacted_covars) <- paste0(treat_name, ':', colnames(demeaned_covars))
+  #n_int_covar_cols <- n_covars * (n_treat_cols + has_intercept)
+  n_int_covar_cols <- n_covars * (n_treat_cols)
+  interacted_covars <- matrix(0, nrow = n, ncol = n_int_covar_cols)
+  interacted_covars_names <- character(n_int_covar_cols)
+  for (i in 1:n_covars) {
+
+    covar_name <- colnames(demeaned_covars)[i]
 
-  # Interact with treatment
-  X <- cbind(design_matrix[, attr(design_matrix, "assign") <= 1, drop = F],
-             demeaned_covars,
-             interacted_covars)
+    cols <- (i - 1) * n_treat_cols + (1:n_treat_cols)
+    interacted_covars[, cols] <-  apply(treatment, 2, `*`, demeaned_covars[, i])
```

Comment:
``` treatment * demeaned_covars[,i] ``` would be easier to read?

Reply:
It doesn't do the correct operation if treatment has multiple columns.

Reply:
oy

Reply:
what do you mean? Am I missing something?
```

> X <- matrix(1:4, 2,2)
> X
     [,1] [,2]
[1,]    1    3
[2,]    2    4
> X * 1:2
     [,1] [,2]
[1,]    1    3
[2,]    4    8
> apply(X, 2, `*`, 1:2)
     [,1] [,2]
[1,]    1    3
[2,]    4    8
```

Reply:
Hmmm, maybe at some point I had it multiplying by all of demeaned_covars instead of one column at a time. Your suggestion should work.